test: added catch statement in lifecycle-operator integration tests

UtkarshUmre opened the pull request:
<!-- PLEASE USE THE TEMPLATE SECTION THAT IS APPLICABLE FOR YOUR CONTRIBUTION -->

<!-- CODE SECTION -->
<!-- USE THIS FOR CODE CONTRIBUTIONS -->

## Description

This issue is about adding catch statements to integration tests for improved error handling.



Fixes #2698 





## Checklist

- [x] My PR fulfills the Definition of Done of the corresponding issue and not more (or parts if the issue is separated
  into multiple PRs)
- [x] I used descriptive commit messages to help reviewers understand my thought process
- [x] I signed off all my commits according to the Developer Certificate of Origin (DCO)
  see [Contribution Guide](https://lifecycle.keptn.sh/contribute/docs/contribution-guidelines)
- [x] My PR title is formatted according to the semantic PR conventions described in
  the [Contribution Guide](https://lifecycle.keptn.sh/contribute/docs/contribution-guidelines)
- [x] My code follows the style guidelines of this project (golangci-lint passes, YAMLLint passes)
- [ ] I regenerated the auto-generated docs for Helm and the CRD documentation (if applicable)
- [x] I have performed a self-review of my code
- [ ] I have made corresponding changes to the documentation (if needed)
- [x] My changes result in all-green PR checks (first-time contributors need to ask a maintainer to approve their test runs)
- [x] New and existing unit and integration tests pass locally with my changes

<!-- DOCS SECTION -->
<!-- USE THIS FOR DOCS CONTRIBUTIONS -->

## Summary

In this pull request, I've implemented catch statements within the integration test. I've conducted thorough local testing to confirm their effectiveness in resolving the issue.

odubajDT commented:
Hi @UtkarshUmre , thanks for your PR. Did you check if the things you are catching the the tests are valid things to check if the test fails? All `catch` blocks are the same and in some cases (for example api conversions) it does not make sense to describe app pod (because it does not exist) and as well keptn resources. 

Please if possible go through each test and re-evaluate if it's needed.

Thank you!

UtkarshUmre commented:
Hey @odubajDT I've re-evaluated the catch statements & addressed the issues with api conversions. i've made the necessary corrections, Could you please review it

odubajDT commented on test/chainsaw/integration/TTLSecondsAfterFinished-in-jobs/chainsaw-test.yaml:
```diff
@@ -11,3 +11,14 @@ spec:
             file: 00-install.yaml
         - assert:
             file: 00-assert.yaml
+      catch:
```
Are you sure that keptnworkloadinstance and keptnapp are needed here? AFAIK they are not created in this test (please check 00-install.yaml). Also metrics-operator is not used here

odubajDT commented:
> Hey @odubajDT I've re-evaluated the catch statements & addressed the issues with api conversions. i've made the necessary corrections, Could you please review it

Hi @UtkarshUmre , thank you for the adaptations. From what I see there are still a lot of information in the catch statement that are not needed.

My tip would be: Try to go through the tests and look at the install and assert files -> from the content (what is applied to the cluster and what is checked/asserted) you should be able to see what resources make sense to be retrieved in case the test fails

odubajDT commented:
Hi @UtkarshUmre any updates on this?

UtkarshUmre commented:
Hey @odubajDT  I'm tied up with some work at the moment, but I'll revisit the PR issues in a couple of days

odubajDT commented:
@UtkarshUmre any updates on this please?

UtkarshUmre commented:
@odubajDT Half of the file's done, but the rest needs another look. I'll update the PR soon.

odubajDT commented:
Closing due to inactivity
